Sources/DataPipeline/DataPipeline.swift

```diff
@@ -157,7 +157,7 @@ public extension DataPipeline {
         implementation?.processRegisterDeviceFromBGQ(identifier: identifier, token: token, timestamp: timestamp, attributes: attributes)
     }
 
-    func processPushMetricsFromBGQ(token: String, event: Metric, deliveryId: String, timestamp: String, metaData: [String: Any] = [:]) {
+    func processPushMetricsFromBGQ(token: String?, event: String, deliveryId: String, timestamp: String, metaData: [String: Any] = [:]) {
```

Comment:
To use common method for push metrics and in app metrics.

Reply:
I understand having a common method may provide value, but I also think non-optional parameters and strong data types also provide value. 

I am open to ideas from others on the team, but I am leaning towards reverting back to the original method signature and introducing another method for in-app.

Reply:
since this is just for migration and one time use, i am not super opinionated on anything, I'll be okay either way,

Reply:
Introducing another method only for in-app would kinda be a duplicate method with similar functionality. Also, as Shahroz mentioned, this method is specific to migration and is only one time run method.